### ISSUE TITLE
CTS failure in nightly

### ISSUE THREAD

**Issue by sarnex** [intel/llvm#14517]
### Describe the bug

```
  build-cts/bin/test_accessor
  Wed Jul 10 04:22:26 UTC 2024
  Randomness seeded to: 2079819610
  
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  The SYCL 2020 Conformance Test Suite is a Catch2 v3.2.1 host application.
  Run with -? for options
  
  -------------------------------------------------------------------------------
  LegacyRandomAccessIterator requirement verification for sycl::accessor iterator
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:121
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Provided iterator not satisfy to LegacyOutputIterator requirement.
    iterator_traits::reference must be const T& or const T&&.
  
  -------------------------------------------------------------------------------
  LegacyRandomAccessIterator requirement verification for sycl::accessor const
  iterator
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:137
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  -------------------------------------------------------------------------------
  LegacyRandomAccessIterator requirement verification for sycl::accessor reverse
  iterator
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:153
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Provided iterator not satisfy to LegacyOutputIterator requirement.
    iterator_traits::reference must be const T& or const T&&.
  
  -------------------------------------------------------------------------------
  LegacyRandomAccessIterator requirement verification for sycl::accessor const
  reverse iterator
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:169
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/accessor/accessor_iterator_requirement.cpp:47: FAILED:
  explicitly with message:
    Iterator must be swappable.
  
  ===============================================================================
  test cases:   1138 |   1134 passed |  4 failed
  assertions: 109211 | 109201 passed | 10 failed
  
```

https://github.com/intel/llvm/actions/runs/9867495007/job/27248337477

**Comment by sarnex** [comment]
@aelovikov-intel Any idea on candidate changes to have caused this or should I just bisect?

**Comment by aelovikov-intel** [comment]
No idea...

**Comment by sarnex** [comment]
```
3800814750da51d6da852ce404bde91e1dbe02b8 is the first bad commit
commit 3800814750da51d6da852ce404bde91e1dbe02b8
Author: Artur Gainullin <artur.gainullin@intel.com>

Date:   Tue Jul 9 08:28:55 2024 -0700

    [SYCL] Key/Value sorting with fixed-size private array input (#14399)

    Implementation of sort_key_value_over_group APIs from

    https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#functions-with-fixed-size-arrays

```

@againull Can you take a look at this? Let me know if you need help reproducing

**Comment by againull** [comment]
> ```
> 3800814750da51d6da852ce404bde91e1dbe02b8 is the first bad commit
> commit 3800814750da51d6da852ce404bde91e1dbe02b8
> Author: Artur Gainullin <artur.gainullin@intel.com>
> 
> Date:   Tue Jul 9 08:28:55 2024 -0700
> 
>     [SYCL] Key/Value sorting with fixed-size private array input (#14399)
> 
>     Implementation of sort_key_value_over_group APIs from
> 
>     https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#functions-with-fixed-size-arrays
> ```
> 
> @againull Can you take a look at this? Let me know if you need help reproducing

Yes, working on it.
